### DISCUSSION TITLE
Iceberg storage caching on ephemeral disks

### DISCUSSION THREAD

**Discussion by mayankasthana** [trinodb/trino#18988] [Ideas]
We use iceberg connector with data stored in GCS. Trino runs on GCP VMs. GCP provides [local-ssd](https://cloud.google.com/local-ssd) which are ephemeral directly attached NVME SSD disks. These would be great for caching.

However, icerberg catalog doesn't provide [storage caching](https://trino.io/docs/current/connector/hive-caching.html) that is present in hive connector.

What would be a good way to do some caching? Can we add this functionality to trino on iceberg-connector? And perhaps other connectors?

There is also alluxio that enables [caching for iceberg on trino](https://www.alluxio.io/blog/get-started-with-trino-and-alluxio-in-5-minutes/), but I am unsure how it will work with ephemeral storage.
I am also unsure if hive connector storage caching works well with ephemeral disks or not.

**Comment by raunaqmorarka** [comment]
This is implemented now https://trino.io/docs/current/object-storage/file-system-cache.html
